### PR TITLE
added git support for home-assistant

### PR DESCRIPTION
**Pull request by runningman84**
#### What this PR does / why we need it:
This pull request adds git sync support

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by billimek** [comment]
@runningman84 can you please do the following for this one?

1. Bump the chart version again (current is `0.6.0`) to something like `0.7.0`?
1. 'sign' the DCO thing (see [this](https://github.com/helm/charts/pull/12624/checks?check_run_id=87190881))
1. Document the new fields added to `values.yaml` in the `README.md`
1. Maybe add something additional to the README explaining about the new usage and the requirement to separately create a secret to be consumed?

**Comment by runningman84** [comment]
@billimek I hope everything looks fine now...

**Review by billimek — COMMENTED** [review]
/ok-to-test

File: stable/home-assistant/templates/deployment.yaml [hunk 1]
```diff
@@ -53,6 +74,13 @@ spec:
             failureThreshold: 5
             timeoutSeconds: 5
           env:
+            {{- if .Values.configurator.hassApiPassword }}
```
Comment (billimek) [inline stable/home-assistant/templates/deployment.yaml]: Is there a reason we need to now set the `HC_HASS_API_PASSWORD` env variable inside the home-assistant container in addition to the configuration container?

**Comment by billimek** [comment]
@runningman84 this looks great, thank you again for submitting this change.  Just one question about the `HC_HASS_API_PASSWORD` env variable in the home-assistant container (see code comment in-line)

File: stable/home-assistant/templates/deployment.yaml (hunk 1 above)
Comment (runningman84) [reply stable/home-assistant/templates/deployment.yaml]: Actually I have no idea. I guess this is not really needed. Do you want me to delete it?

Comment (billimek) [reply stable/home-assistant/templates/deployment.yaml]: Sure, go for it!  It may have accidentally been introduced as part of the change.

**Pull request closed without merging**